src/driver/sdfmt.d

```diff
@@ -66,15 +66,18 @@ int main(string[] args) {
 		}
 
 		import format.writer;
-		auto o = chunks.write(conf);
+		const input = base.getFullPosition(context).getSource().getContent();
+		const output = chunks.write(conf);
 
 		if (inPlace) {
-			// TODO: Only write if the file changed.
-			import std.file;
-			filename.write(o);
+			import std.string : cmp;
+			if (!cmp(input, output)) {
+				import std.file : write;
+				filename.write(output);
+			}
 		} else {
-			import std.stdio;
-			write(o);
+			import std.stdio : write;
```

Comment:
I don't think this add much value. I don't know where this pattern comes from, and what problem it is trying to solve, but it just gets very verbose very quickly.

Reply:
I think its nice to set a standard for the rest of the project so you don't end up with 
```
import dmd.access;
import dmd.aggregate;
import dmd.aliasthis;
import dmd.arrayop;
import dmd.arraytypes;
import dmd.attrib;
import dmd.astcodegen;
import dmd.astenums;
import dmd.canthrow;
import dmd.chkformat;
import dmd.ctorflow;
import dmd.dscope;
import dmd.dsymbol;
import dmd.declaration;
import dmd.dclass;
import dmd.dcast;
import dmd.delegatize;
import dmd.denum;
import dmd.dimport;
import dmd.dinterpret;
import dmd.dmangle;
import dmd.dmodule;
import dmd.dstruct;
import dmd.dsymbolsem;
import dmd.dtemplate;
import dmd.errors;
import dmd.escape;
import dmd.expression;
import dmd.file_manager;
import dmd.func;
import dmd.globals;
import dmd.hdrgen;
import dmd.id;
import dmd.identifier;
import dmd.imphint;
import dmd.importc;
import dmd.init;
import dmd.initsem;
import dmd.inline;
import dmd.intrange;
import dmd.location;
import dmd.mtype;
import dmd.mustuse;
import dmd.nspace;
import dmd.opover;
import dmd.optimize;
import dmd.parse;
import dmd.printast;
import dmd.root.array;
import dmd.root.ctfloat;
import dmd.root.filename;
import dmd.common.outbuffer;
import dmd.root.rootobject;
import dmd.root.string;
import dmd.root.utf;
import dmd.semantic2;
import dmd.semantic3;
import dmd.sideeffect;
import dmd.safe;
import dmd.target;
import dmd.tokens;
import dmd.traits;
import dmd.typesem;
import dmd.typinf;
import dmd.utils;
import dmd.visitor;
```
:)

Reply:
If you actually use a bunch of stuff then import it but it genuinely is quite effective as a way of avoiding bullshit imports (e.g. Walter insists on every dmd module importing a bunch of spurious nonsense so its easier to inject printf)

Reply:
Just import the module you need, at the scope you need. Done.